packages/chain-adapters/src/cosmossdk/cosmos/CosmosChainAdapter.ts

```diff
@@ -134,26 +134,53 @@ export class ChainAdapter
         addressNList,
         tx: utx,
         chain_id: caip2.ChainReference.CosmosHubMainnet,
-        account_number: '',
-        sequence: ''
+        account_number: account.chainSpecific.accountNumber,
+        sequence: account.chainSpecific.sequence
       }
       return { txToSign }
     } catch (err) {
       return ErrorHandler(err)
     }
   }
 
-  async getFeeData(
-    /* eslint-disable-next-line @typescript-eslint/no-unused-vars -- Disable no-unused-vars lint rule for unimplemented methods */
-    input: Partial<chainAdapters.GetFeeDataInput<ChainTypes.Cosmos>>
-  ): Promise<chainAdapters.FeeDataEstimate<ChainTypes.Cosmos>> {
-    throw new Error('Method not implemented.')
+  async getFeeData({
```

Comment:
Hard coding the fees here because we don't have a way to dynamically get fee data without sending a singed transaction first. This would break the current send flow by making the user sign a transaction first, then get the fee data, then sign again with the correct fee data. The fee values are arbitrary, but we wanted to keep the fees low and similar to other wallet fees. Any feedback is welcome.

Reply:
@DaoDev44 @willyogo @DiggyDiggy2 how do other cosmos apps approach this? is there really no way to get fee data without a signed message?

Reply:
@0xdef1cafe After discussing with @kaladinlight, this is probably the best path in the sort term, however, unchained can be modified to take similar `getFeeData` arguments as we do with ETH and serialize it to a format that the gaia node can accept, but that might be a bigger lift and can be done in the second round of cosmos improvements.